File: app/src/main/java/ru/profapp/RanobeReader/Helpers/StringHelper.java
package ru.profapp.RanobeReader.Helpers;

import org.jsoup.Jsoup;

/**
 * Created by Ruslan on 08.02.2018.
 */

public class StringHelper {

    private static volatile StringHelper instance;

    private StringHelper() {
    }

    public static StringHelper getInstance() {
        if (instance == null) {
            synchronized (StringHelper.class) {
                if (instance == null) {
                    instance = new StringHelper();
                }
            }
        }
        return instance;
    }

    public String cleanJson(String string) {

        string = string.substring(0, string.lastIndexOf("}") + 1);
        return string;
    }

    public String removeTags(String string) {
        return Jsoup.parse(string).text();
    }

    public String cleanAdditionalInfo(String string) {
       return Jsoup.parse(string).text();
    }
}

File: app/src/main/java/ru/profapp/RanobeReader/Helpers/StringHelper.kt
package ru.profapp.RanobeReader.Helpers

import android.text.Html

/**
 * Created by Ruslan on 08.02.2018.
 */
class StringHelper {
    companion object {

        fun cleanJson(string: String): String {
            return string.substring(0, string.lastIndexOf("}") + 1)
        }


        fun removeTags(string: String): String {
            return Html.fromHtml(string).toString().replace("\n\n", "\n")
        }


        fun CleanString(ranobeUrl: String): String {
            return ranobeUrl.replace("[^a-zA-Z0-9]".toRegex(), "")
        }

    }
}
